### PR TITLE
Merge v8 into main

### PR DESCRIPTION
**Pull request by buenaflor**
merges the v8 branch into main from which we then can do a 8.0.0 release

#skip-changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -117,6 +139,40 @@
   - [changelog](https://github.com/getsentry/sentry-cocoa/blob/main/CHANGELOG.md#8200)
   - [diff](https://github.com/getsentry/sentry-cocoa/compare/8.19.0...8.20.0)
 
+## 8.0.0-beta.2
```
Comment (ueman) [inline CHANGELOG.md:146]: This is at the wrong place in the changelog

Comment (buenaflor) [reply CHANGELOG.md:146]: What do you mean?

Comment (ueman) [reply CHANGELOG.md:146]: The v8 things should be at the top, right? That's not the case right now.

Comment (buenaflor) [reply CHANGELOG.md:146]: We discussed this and we'll keep the release order so beta should be fine where it is now

**Review by kahest — APPROVED** [review]
🚀

**Pull request merged**